webrender/src/segment.rs

```diff
@@ -413,140 +423,152 @@ impl SegmentBuilder {
             None => return,
         };
 
-        // First, filter out any items that don't intersect
-        // with the visible bounding rect.
-        self.items.retain(|item| item.rect.intersects(&bounding_rect));
+        if self.has_interesting_clips {
```

Comment:
nit: I have a slight preference for avoiding right drift by doing:

```rust
if early_out_cond {
    // early out code
}

// most of the stuff
```

I think that avoiding lots of nested scope makes the code generally easier to read and less intimidating.

Reply:
In general I agree with you. I didn't do that in this case because there is a piece of code at the end of the method (that resets the initialization state) that much run regardless of which branch is taken. Although, having said that, I just realized that can be moved above the early out in this case.

Reply:
Pushed an update to resolve this.